pipeline/02-assess.R

```diff
@@ -65,9 +87,54 @@ message("Performing post-modeling adjustments")
 ## 3.1. Multicards -------------------------------------------------------------
 message("Fixing multicard PINs")
 
+# Re-add dropped multicard rows, distributing total predicted FMV
+# across each original card by share of bldg sqft
+
+# Identify the dropped multi-card rows
+df_multi_card_dropped <- df_assessment_data %>%
+  filter(ind_pin_is_multicard) %>%
+  anti_join(
+    df_multi_card_kept %>% select(meta_pin, meta_card_num),
+    by = c("meta_pin", "meta_card_num")
+  )
+
+# Combine kept predicted rows with 'dropped' rows
+df_multi_card_combined <- assessment_card_data_pred %>%
+  filter(ind_pin_is_multicard) %>%
+  bind_rows(
+    df_multi_card_dropped %>% mutate(pred_card_initial_fmv = NA_real_)
+  )
+
+# For each PIN, distribute total predicted FMV proportionally
+df_multi_card_final <- df_multi_card_combined %>%
+  group_by(meta_pin) %>%
+  mutate(
+    total_fmv = sum(pred_card_initial_fmv, na.rm = TRUE),
+    total_bldg_sf_pin = sum(char_bldg_sf, na.rm = TRUE),
+    share_bldg_sf = char_bldg_sf / total_bldg_sf_pin,
+    pred_card_initial_fmv = total_fmv * share_bldg_sf
```

Comment:
There is a small amount of multi-card pins with a card having a sqft of 0, which results in the price for the card being 0. I'm not sure how to handle this. We could just make some fixed cut if that is the case. Open to ideas